develop: aarch64 + configure compilation issue recently introduced.

My last working build 17 days ago:  https://github.com/userdocs/qbittorrent-nox-static/actions/runs/1775536375

Broken builds from today : https://github.com/userdocs/qbittorrent-nox-static/runs/5233671039

I am guessing this? https://github.com/zlib-ng/zlib-ng/commit/9146bd472c23a501725d275f8b7edec93c072466#diff-90d08e583c4c9c6f391b2ae90f819f600a6326928ea9512c9e0c6d98e9f29ac2

I will try to test and confirm but the dates and commit type add up.

```bash
Checking for __builtin_ctzll ... Yes.
./configure: line 1828: test: -eq: unary operator expected
ARCH: armv8-a+crc+simd
Using arch directory: arch/arm
Architecture-specific static object files: arm_features.o crc32_acle.o insert_string_acle.o adler32_neon.o chunkset_neon.o slide_hash_neon.o
Architecture-specific shared object files: arm_features.lo crc32_acle.lo insert_string_acle.lo adler32_neon.lo chunkset_neon.lo slide_hash_neon.lo
aarch64-linux-musl-gcc -O2  -std=c11 -Wall -DNDEBUG -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DZLIB_COMPAT -DWITH_GZFILEOP -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_BUILTIN_CTZ -DHAVE_BUILTIN_CTZLL -DARM_FEATURES -DARM_ACLE_CRC_HASH -DARM_NEON_ADLER32 -DARM_NEON_CHUNKSET -DARM_NEON_SLIDEHASH -DUNALIGNED_OK -DUNALIGNED64_OK -I/root/qbt-build/zlib -c -o test_adler32.o /root/qbt-build/zlib/test/test_adler32.c
aarch64-linux-musl-gcc -O2  -std=c11 -Wall -DNDEBUG -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DZLIB_COMPAT -DWITH_GZFILEOP -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_BUILTIN_CTZ -DHAVE_BUILTIN_CTZLL -DARM_FEATURES -DARM_ACLE_CRC_HASH -DARM_NEON_ADLER32 -DARM_NEON_CHUNKSET -DARM_NEON_SLIDEHASH -DUNALIGNED_OK -DUNALIGNED64_OK -DWITH_GZFILEOP -I/root/qbt-build/zlib -c -o gzlib.o /root/qbt-build/zlib/gzlib.c
aarch64-linux-musl-gcc -O2  -std=c11 -Wall -DNDEBUG -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DZLIB_COMPAT -DWITH_GZFILEOP -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_BUILTIN_CTZ -DHAVE_BUILTIN_CTZLL -DARM_FEATURES -DARM_ACLE_CRC_HASH -DARM_NEON_ADLER32 -DARM_NEON_CHUNKSET -DARM_NEON_SLIDEHASH -DUNALIGNED_OK -DUNALIGNED64_OK -DWITH_GZFILEOP -I/root/qbt-build/zlib -c -o gzread.o /root/qbt-build/zlib/gzread.c
aarch64-linux-musl-gcc -O2  -std=c11 -Wall -DNDEBUG -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DZLIB_COMPAT -DWITH_GZFILEOP -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_BUILTIN_CTZ -DHAVE_BUILTIN_CTZLL -DARM_FEATURES -DARM_ACLE_CRC_HASH -DARM_NEON_ADLER32 -DARM_NEON_CHUNKSET -DARM_NEON_SLIDEHASH -DUNALIGNED_OK -DUNALIGNED64_OK -DWITH_GZFILEOP -I/root/qbt-build/zlib -c -o gzwrite.o /root/qbt-build/zlib/gzwrite.c
aarch64-linux-musl-gcc -O2  -std=c11 -Wall -DNDEBUG -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DZLIB_COMPAT -DWITH_GZFILEOP -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_BUILTIN_CTZ -DHAVE_BUILTIN_CTZLL -DARM_FEATURES -DARM_ACLE_CRC_HASH -DARM_NEON_ADLER32 -DARM_NEON_CHUNKSET -DARM_NEON_SLIDEHASH -DUNALIGNED_OK -DUNALIGNED64_OK -I/root/qbt-build/zlib -c -o adler32.o /root/qbt-build/zlib/adler32.c
aarch64-linux-musl-gcc -O2  -std=c11 -Wall -DNDEBUG -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DZLIB_COMPAT -DWITH_GZFILEOP -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_BUILTIN_CTZ -DHAVE_BUILTIN_CTZLL -DARM_FEATURES -DARM_ACLE_CRC_HASH -DARM_NEON_ADLER32 -DARM_NEON_CHUNKSET -DARM_NEON_SLIDEHASH -DUNALIGNED_OK -DUNALIGNED64_OK -I/root/qbt-build/zlib -c -o chunkset.o /root/qbt-build/zlib/chunkset.c
aarch64-linux-musl-gcc -O2  -std=c11 -Wall -DNDEBUG -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DZLIB_COMPAT -DWITH_GZFILEOP -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_BUILTIN_CTZ -DHAVE_BUILTIN_CTZLL -DARM_FEATURES -DARM_ACLE_CRC_HASH -DARM_NEON_ADLER32 -DARM_NEON_CHUNKSET -DARM_NEON_SLIDEHASH -DUNALIGNED_OK -DUNALIGNED64_OK -I/root/qbt-build/zlib -c -o compare256.o /root/qbt-build/zlib/compare256.c
aarch64-linux-musl-gcc -O2  -std=c11 -Wall -DNDEBUG -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DZLIB_COMPAT -DWITH_GZFILEOP -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_BUILTIN_CTZ -DHAVE_BUILTIN_CTZLL -DARM_FEATURES -DARM_ACLE_CRC_HASH -DARM_NEON_ADLER32 -DARM_NEON_CHUNKSET -DARM_NEON_SLIDEHASH -DUNALIGNED_OK -DUNALIGNED64_OK -I/root/qbt-build/zlib -c -o compress.o /root/qbt-build/zlib/compress.c
aarch64-linux-musl-gcc -O2  -std=c11 -Wall -DNDEBUG -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DZLIB_COMPAT -DWITH_GZFILEOP -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_BUILTIN_CTZ -DHAVE_BUILTIN_CTZLL -DARM_FEATURES -DARM_ACLE_CRC_HASH -DARM_NEON_ADLER32 -DARM_NEON_CHUNKSET -DARM_NEON_SLIDEHASH -DUNALIGNED_OK -DUNALIGNED64_OK -I/root/qbt-build/zlib -c -o cpu_features.o /root/qbt-build/zlib/cpu_features.c
aarch64-linux-musl-gcc -O2  -std=c11 -Wall -DNDEBUG -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DZLIB_COMPAT -DWITH_GZFILEOP -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_BUILTIN_CTZ -DHAVE_BUILTIN_CTZLL -DARM_FEATURES -DARM_ACLE_CRC_HASH -DARM_NEON_ADLER32 -DARM_NEON_CHUNKSET -DARM_NEON_SLIDEHASH -DUNALIGNED_OK -DUNALIGNED64_OK -I/root/qbt-build/zlib -c -o crc32.o /root/qbt-build/zlib/crc32.c
aarch64-linux-musl-gcc -O2  -std=c11 -Wall -DNDEBUG -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DZLIB_COMPAT -DWITH_GZFILEOP -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_BUILTIN_CTZ -DHAVE_BUILTIN_CTZLL -DARM_FEATURES -DARM_ACLE_CRC_HASH -DARM_NEON_ADLER32 -DARM_NEON_CHUNKSET -DARM_NEON_SLIDEHASH -DUNALIGNED_OK -DUNALIGNED64_OK -I/root/qbt-build/zlib -c -o crc32_comb.o /root/qbt-build/zlib/crc32_comb.c
In file included from /root/qbt-build/zlib/compare256.c:7:
/root/qbt-build/zlib/fallback_builtins.h:87:28: error: conflicting types for ‘vld1q_u16_x4’; have ‘uint16x8x4_t(uint16_t *)’ {aka ‘uint16x8x4_t(short unsigned int *)’}
   87 | static inline uint16x8x4_t vld1q_u16_x4(uint16_t *a) {
      |                            ^~~~~~~~~~~~
In file included from /root/qbt-build/zlib/fallback_builtins.h:84,
                 from /root/qbt-build/zlib/compare256.c:7:
/root/qbt-build/lib/gcc/aarch64-linux-musl/11.2.1/include/arm_neon.h:16589:1: note: previous definition of ‘vld1q_u16_x4’ with type ‘uint16x8x4_t(const uint16_t *)’ {aka ‘uint16x8x4_t(const short unsigned int *)’}
16589 | vld1q_u16_x4 (const uint16_t *__a)
      | ^~~~~~~~~~~~
aarch64-linux-musl-gcc -O2  -std=c11 -Wall -DNDEBUG -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DZLIB_COMPAT -DWITH_GZFILEOP -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_BUILTIN_CTZ -DHAVE_BUILTIN_CTZLL -DARM_FEATURES -DARM_ACLE_CRC_HASH -DARM_NEON_ADLER32 -DARM_NEON_CHUNKSET -DARM_NEON_SLIDEHASH -DUNALIGNED_OK -DUNALIGNED64_OK -I/root/qbt-build/zlib -c -o crc32_fold.o /root/qbt-build/zlib/crc32_fold.c
make: *** [Makefile:278: compare256.o] Error 1
make: *** Waiting for unfinished jobs....
```

--- a/configure
+++ b/configure
@@ -1812,6 +1812,8 @@ EOF
         fi
 
         if test $without_optimizations -eq 0; then
+            check_neon_ld4_intrinsics
+
             CFLAGS="${CFLAGS} -DARM_FEATURES"
             SFLAGS="${SFLAGS} -DARM_FEATURES"
             ARCH_STATIC_OBJS="${ARCH_STATIC_OBJS} armfeature.o"

